docs/pages/_app.js

```diff
@@ -59,7 +64,6 @@ ponyfillGlobal.muiDocConfig = {
       '@mui/x-date-pickers-pro': getMuiPackageVersion('x-date-pickers-pro', muiCommitRef),
       '@mui/x-charts': getMuiPackageVersion('x-charts', muiCommitRef),
       '@mui/x-tree-view': getMuiPackageVersion('x-tree-view', muiCommitRef),
-      'date-fns': 'latest',
```

Comment:
Defining the version did not work, because this would override it:

https://github.com/mui/material-ui/blob/8be7ac66b99059bf2c2edcca9cf15981d389db6b/docs/src/modules/sandbox/Dependencies.ts#L157

WDYT @flaviendelangle, does the proposal make sense or shall we refactor the code in `monorepo`?
But in that case, we'd be blocked for the moment, because we currently can't bump `monorepo`. 🙈

Reply:
How long do we think we will be blocked from bumping the monorepo?

Having `latest` hardcoded is really not a good pattern :grimacing:

Reply:
> How long do we think we will be blocked from bumping the monorepo?

Given the complexity of https://github.com/mui/mui-x/pull/11303 I'd bet that it is not happening this year. 😃 
So, at least 2+ weeks of a few broken demos in codesandbox.


> Having `latest` hardcoded is really not a good pattern 😬

I don't disagree. It's error-prone, like we see now. 🙈 
But I'd say that in such case, all of it should be controlled on the consuming side.
Now we have a bunch of logic in monorepo, some of it overrides whatever we specify here... 🤷

Reply:
I would be totally fine if you took the current commit of the monorepo on your repo and just did the date-fns fix 
That's not clean, but this is a very bad regression on our doc :grimacing: 

> But I'd say that in such case, all of it should be controlled on the consuming side.

Yes you're right
`latest` is bad AND having to go to the monorepo to fix it is bad...

Reply:
@flaviendelangle WDYT about the updated solution with fixes originating from `@mui/monorepo`? 🤔